pkg/operator/operator_config.go

```diff
@@ -58,8 +74,9 @@ func setupOperatorConfigControllers(op *Operator) error {
 
 // operatorConfigReconciler reconciles the OperatorConfig CRD.
 type operatorConfigReconciler struct {
-	client client.Client
-	opts   Options
+	client     client.Client
+	opts       Options
+	secretData map[string][]byte
```

Comment:
Seems like this can just be returned by `ensureRuleEvaluatorConfig` and then passed into `ensureRuleEvaluatorSecrets` instead of putting it into the struct. Makes the data flow more obvious, too.